File: 25_MergeSortedLists/MergeSortedLists.cpp
#include <iostream>
#include <vector>
#include <string>
#include <deque>
#include <set>
#include <map>
#include <queue>
#include <stack>
#include <algorithm>
#include <cstring>
using namespace std;

struct ListNode
{
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution
{
  public:
    ListNode *Merge(ListNode *pHead1, ListNode *pHead2)
    {
        if (!pHead1)
            return pHead2;
        else if (!pHead2)
            return pHead1;
        ListNode *mergedHead = nullptr;
        if (pHead1->val < pHead2->val)
        {
            mergedHead = pHead1;
            mergedHead->next = Merge(pHead1->next, pHead2);
        }
        else
        {
            mergedHead = pHead2;
            mergedHead->next = Merge(pHead1, pHead2->next);
        }
        return mergedHead;
    }
};

File: 25_MergeSortedLists/MergeSortedLists.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    # 返回合并后列表
    def Merge(self, pHead1, pHead2):
        # write code here
        if pHead1 is None:
            return pHead2
        elif pHead2 is None:
            return pHead1
        mergedHead = None
        if pHead1.val < pHead2.val:
            mergedHead = pHead1
            mergedHead.next = self.Merge(pHead1.next, pHead2)
        else:
            mergedHead = pHead2
            mergedHead.next = self.Merge(pHead1, pHead2.next)
        return mergedHead
